### DISCUSSION TITLE
Sensitive Data Exposure

### DISCUSSION THREAD

**Discussion by b-codex** [vercel/community#6020] [Help]
### Summary

when my hosted application went through a security test, some sensitive files and certificates were exposed and those files should not be visible to the public. These files have naming based on the website and the IP address of the website with extensions such ass .war, .tar, .jks, .egg and .cert.

How can I solve this?

### Example

Not advised to share

### Steps to Reproduce

I don't exactly know the steps to reproduce this since the security test was handled by an external organization.

### Image I received for the security issue
![Screenshot 2024-03-06 050358](https://github.com/vercel/community/assets/61845339/dfa25d7c-ffd0-4b5f-a0d0-9e620a82564d)

**Comment by amyegan** [comment]
Hey @b-codex. Obviously I can't do a full code review in this situation, as you have sensitive files to protect. But I can say that anything you want to keep hidden should generally be kept on the server side and any client side files should generally be considered public. I hope that helps 🙏

**Reply by b-codex** [reply]
the codebase is written in NextJS and there is no server. the database used is firebase and some of its services for other features of the project.

**Reply by b-codex** [reply]
i have updated the post and added an image i received from them. hopefully it'll help.
